sources/common/toddl.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"strconv"
+	"unicode"
+
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/schema"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+)
+
+type BaseToDdl interface {
```

Comment:
Do you think if it makes sense to rename ToDdl to ToSpannerDdl? We don't need to define a common interface for destination types yet, but naming this way will make it easier to add other destination targets.

Reply:
I was looking at this interface as a common interface for all type mappings. The differentiation is by methods like ToSpannerType. We can add more methods when we add more target databases. If the number of targets becomes too many to manage in this manner, we can re-evaluate